crawl-ref/source/directn.h

```diff
@@ -106,6 +106,15 @@ struct direction_chooser_args
 
 };
 
+struct dist_sorter
+{
+    coord_def pos;
+    bool operator()(const actor* a, const actor* b)
+    {
+        return a->pos().distance_from(pos) > b->pos().distance_from(pos);
```

Comment:
Note the comparison `>`: this sorter sorts actors from farthest to nearest order. If you want to re-factor and generalize to also use this to have potions of attraction sort by nearest, you should do one of two things:

-  add a boolean field to the struct and make the comparison be nearest-first or farthest first
- create two structs, one for each behavior

Regardless of which one is chosen, also add a comment to explain the two fields and the structs purpose.